core/src/main/scala/stryker4s/report/MutantRunReporter.scala

```diff
@@ -0,0 +1,12 @@
+package stryker4s.report
+
+import stryker4s.model.{Mutant, MutantRunResult, MutantRunResults}
+
+trait MutantRunReporter {
```

Comment:
I find this interface very strange, it feels like this was designed with only the console reporter in mind. 
When looking at the HTML reporter this interface doesn't make sense anymore. 

There are two different types of reporters, in my opinion, one for reporting the final result of the complete mutation run. and a second type would be a progress reporter, this progress reporter will report the current status etc. This progress reporter is something we could automatically turn on untill the user maybe turns something like a silent mode on or set logging levels accordingly. 

What are your toughts about this @hugo-vrijswijk ?

Reply:
I've refactored the reporter to two separate traits: `ProgressReporter`, and `FinishedRunReporter`. Any reporter can implement the traits and the `CombinedReporter` will call it if it is implemented